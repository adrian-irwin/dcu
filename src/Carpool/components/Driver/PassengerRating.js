import React, { useState } from "react";
import { Rating } from "react-native-ratings";
import { Text } from "react-native-paper";
import { BACKEND_URL } from "@env";

// Passenger rating component
const PassengerRating = ({ passenger, index, send }) => {
    // State to store the rating
    const [rating, setRating] = useState(3);

    if (send === true) {
        //api call to send rating to backend
        console.log(`Rating for ${passenger.name} is ${rating}!`);
        fetch(`${BACKEND_URL}/api/passengers/rate`, {
            method: "POST",
            headers: {
                "Content-Type": "application/json"
            },
            body: JSON.stringify({
                passengerID: passenger._id,
                rating: rating
            })
        })
            .then(res => res.json())
            .then(data => {
                console.log(data);
            }
            )
            .catch(err => {
                console.log(err);
            });
    }

    return (
        <>
            <Text style={{ marginVertical: 5, textAlign: "center" }} variant="titleLarge">Rate {passenger.name}!</Text>
            <Rating
                key={index}
                type='star'
                ratingCount={5}
                imageSize={70}
                ratingColor="3F51B5"
                showRating
                startingValue={3}
                onFinishRating={rating => { setRating(rating); }}
            />
        </>
    );
};

export default PassengerRating;
