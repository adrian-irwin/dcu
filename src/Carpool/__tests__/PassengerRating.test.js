import React from "react";
import { render } from "@testing-library/react-native";

import PassengerRating from "../components/Driver/PassengerRating";

describe("PassengerRating", () => {
    it("renders correctly", () => {
        const { toJSON } = render(<PassengerRating passenger={{name: "test", rating: 0}} />);
        expect(toJSON()).toMatchSnapshot();
    });
});
