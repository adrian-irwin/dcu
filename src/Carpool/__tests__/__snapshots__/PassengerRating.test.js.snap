// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PassengerRating renders correctly 1`] = `
Array [
  <Text
    style={
      Array [
        Object {
          "color": "rgba(28, 27, 31, 1)",
          "fontFamily": "System",
          "fontSize": 22,
          "fontWeight": "400",
          "letterSpacing": 0,
          "lineHeight": 28,
        },
        Object {
          "textAlign": "left",
        },
        Object {
          "writingDirection": "ltr",
        },
        Object {
          "marginVertical": 5,
          "textAlign": "center",
        },
      ]
    }
  >
    Rate 
    test
    !
  </Text>,
  <View
    pointerEvents="auto"
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "column",
          "justifyContent": "center",
          "paddingBottom": 5,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "center",
            "paddingBottom": 5,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#34495e",
                "fontFamily": "Trebuchet MS",
                "fontSize": 15,
                "textAlign": "center",
              },
              Object {
                "color": "#f1c40f",
              },
            ]
          }
        >
          Rating: 
        </Text>
        <Text
          style={
            Array [
              Object {
                "fontFamily": "Trebuchet MS",
                "fontSize": 30,
                "textAlign": "center",
              },
              Object {
                "color": "#f1c40f",
              },
            ]
          }
        >
          3
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "#34495e",
                "fontFamily": "Trebuchet MS",
                "fontSize": 18,
                "textAlign": "center",
              },
              Object {
                "color": "#f1c40f",
              },
            ]
          }
        >
          /
          5
        </Text>
      </View>
      <View />
    </View>
    <View
      onMoveShouldSetResponder={[Function]}
      onMoveShouldSetResponderCapture={[Function]}
      onResponderEnd={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderReject={[Function]}
      onResponderRelease={[Function]}
      onResponderStart={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      onStartShouldSetResponderCapture={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "center",
        }
      }
    >
      <View
        onLayout={[Function]}
        style={
          Object {
            "alignItems": "center",
            "bottom": 0,
            "flexDirection": "row",
            "justifyContent": "center",
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <View
          collapsable={false}
          style={
            Object {
              "backgroundColor": "#f1c40f",
              "height": 70,
              "width": 210,
            }
          }
        />
        <View
          collapsable={false}
          style={
            Object {
              "backgroundColor": "white",
              "height": 70,
              "width": 140,
            }
          }
        />
      </View>
      <View>
        <Image
          source={1}
          style={
            Object {
              "height": 70,
              "tintColor": undefined,
              "width": 70,
            }
          }
        />
      </View>
      <View>
        <Image
          source={1}
          style={
            Object {
              "height": 70,
              "tintColor": undefined,
              "width": 70,
            }
          }
        />
      </View>
      <View>
        <Image
          source={1}
          style={
            Object {
              "height": 70,
              "tintColor": undefined,
              "width": 70,
            }
          }
        />
      </View>
      <View>
        <Image
          source={1}
          style={
            Object {
              "height": 70,
              "tintColor": undefined,
              "width": 70,
            }
          }
        />
      </View>
      <View>
        <Image
          source={1}
          style={
            Object {
              "height": 70,
              "tintColor": undefined,
              "width": 70,
            }
          }
        />
      </View>
    </View>
  </View>,
]
`;
